### DISCUSSION TITLE
Different prices for family members discount snippit

### DISCUSSION THREAD

**Discussion by Sp0run** [liedekef/events-made-easy#72] [Show and tell]
I've just been messing around with discounts for memberships. I have a family membership that gives a different price per family member based on age. I have coded a basic way of achieving this using the eme_discount_* filter. If you're hoping to achieve the same here is a good starting point. Hope someone might find use of it :)
```


/* function to apply discounts to bookings */
add_filter('eme_discount_familydiscount', 'family_membership_discount');

function family_membership_discount($booking){
    // reduce the cost of the family membership based on the count of family members

    $adults = 0;
    $kids = 0;
    $today = date('Y-m-d');
    
    //head of accounts age
    $age = calculateAge($_POST['birthdate'],$today);
    
    if($age < 16){
        $kids++;
    }else if ($age > 15){
        $adults++;
    }
    
    
    //if we have family members loop through each and calculate their age
    if(isset($_POST['familymember'])){
        
        foreach($_POST['familymember'] as $fam){
            
            $age = calculateAge($fam['birthdate'], $today);
            if($age < 16){
                $kids++;
            }else if ($age > 15){
                $adults++;
            }
        }
    }

    //calculate total cost per member
    $kid_cost = $kids*20;
    $adult_cost = $adults*25;
    
    //get membership
    $membership = eme_get_membership($_POST['membership_id']);
    
    //calculate discount by removing total adult & child cost from membership cost
    $discount = $membership['properties']['price']-($kid_cost+$adult_cost);
    return $discount;
    
}
```

**Comment by liedekef** [comment]
Small remark: don't use "$booking" as variable, but "$member", it's name is more correct :-)
Also: $member contains the id of the membership, so your call to eme_get_membership could become: `eme_get_membership($member['membership_id']);`  (so you don't need to rely on the $_POST of membership_id here)
Further info: this is based on a discount of type code with the name "familydiscount".
Normally also birthdate could be get from the $person related to the membership (`$person = eme_get_person($member['person_id']);`  and then use $person['birthdate'] ).
But anyway you code should work fine :-)

**Reply by Sp0run** [reply]
Thanks! I'll keep that in mind :)
